components/climate/bang_bang.rst

```diff
@@ -90,6 +97,87 @@ Advanced options:
+
+- As soon as the temperature goes below the *lower* target temperature, ``heat_action`` is called to activate heating.
+- Heating will continue until the temperature reaches the *upper* target temperature, at which point ``idle_action`` is called to stop heating.
+- As soon as the temperature goes above the *upper* target temperature, ``cool_action`` is called to activate cooling.
+- Cooling will continue until the temperature reaches the *lower* target temperature, at which point ``idle_action`` is called to stop cooling.
+
+A single bang-bang controller may work well for systems that only heat or only cool; however, it begins to break down
+when applied to systems that may both heat *and* cool. This is simply because both actions are tied to both set
+points -- that is, the point at which heating stops *is also the point at which cooling begins*. The reverse is also
+true: the point at which cooling stops *is also the point at which heating begins*. Let's look at an example:
+
+Consider a system that both heats and cools. The ``target_temperature_low`` set point is 20 °C while the
+``target_temperature_high`` set point is 22 °C. The sensor reports that the temperature is 19.75 °C, so the controller
+calls ``heat_action`` to activate heating. Heating continues until the temperature reaches ``target_temperature_high``
+(22 °C in this case). Once this temperature is achieved, ``idle_action`` is called to stop heating. *However*, should
+temperature drift even slightly above ``target_temperature_high``--even just a fraction of a degree for a fraction of
+a second--the controller will call ``cool_action`` to begin cooling. Now, cooling will continue until
+``target_temperature_low`` is reached again, but, as before, should the temperature drift even slightly below
+``target_temperature_low`` for even a fraction of a second, ``heat_action`` will be invoked again, and the cycle will
+repeat. It will "ping-pong" between the two set points, potentially forever. Oscillation at the "edges" of the
+hysteresis window, or going past the set point in either direction, should be expected; consider, for example, a heater
+that is turned off after it reaches its set point. The heating element will remain hot (potentially for quite a while),
+and as such will continue to heat the air until the element fully cools down to match the ambient air/room temperature.
+
+The :doc:`thermostat <thermostat>` component differs in that there is hysteresis around *each* set point. For example,
+if the ``target_temperature_low`` set point is 20 °C, and the (default) hysteresis value of 0.5 °C is used,
+``heat_action`` is called at a temperature of 19.5 °C and ``idle_action`` is called at 20.5 °C. If cooling, as defined
+by ``target_temperature_high``, is set to 22 °C, ``cool_action`` would be called at 22.5 °C and ``idle_action`` called
+at 21.5 °C. Again, it is essentially two bang-bang controllers in one.
+
+Great. But what if you do not have a dual-function system?
```

Comment:
```suggestion
Great. But what if you do not have a dual-function system?
```

Hmm, i don't really understand this sentence. If you have a single function system this doc recommends using bang_bang right?

Reply:
Not necessarily -- the recommendation is made at the end in the "Which is Right for Me?" section.
We could remove the line, it's not terribly important. It's just there to help transition the focus from dual-function systems to user interface differences.

Reply:
What if we replaced it with this?:
"Now that we've highlighted the behavioral differences between these two components, let's take a look at the other important difference: User Interface."

Reply:
Yes that would be good (Just was confused when I first read that sentence, I didn't understand what it was referring to)

Reply:
Sweet! I pushed up a change with a modification similar to that. 😄  See https://github.com/esphome/esphome-docs/pull/667/commits/bd55f024fc0e9774cdea5f5cc0fb80ccd8005a49